api/core/app/apps/workflow/app_generator.py

```diff
@@ -445,17 +446,44 @@ def _generate_worker(
         """
 
         with preserve_flask_contexts(flask_app, context_vars=context):
-            try:
-                # workflow app
-                runner = WorkflowAppRunner(
-                    application_generate_entity=application_generate_entity,
-                    queue_manager=queue_manager,
-                    workflow_thread_pool_id=workflow_thread_pool_id,
-                    variable_loader=variable_loader,
+            with Session(db.engine, expire_on_commit=False) as session:
```

Comment:
The logic for querying the workflow and computing `system_user_id` is identical to the implementation in `AdvancedChatAppGenerator`. Consider extracting this duplicated code into a shared helper function or method to improve maintainability and reduce code duplication.